File: ruby/lib/Sikaru/types/trace_import_options_request.rb
# frozen_string_literal: true

module Sikaru
  module Types
    class TraceImportOptionsRequest < Internal::Types::Model
      field :include_feedback, -> { Internal::Types::Boolean }, optional: true, nullable: false, api_name: "includeFeedback"

      field :include_pending_spans, -> { Internal::Types::Boolean }, optional: true, nullable: false, api_name: "includePendingSpans"

      field :infer_agent_boundaries, -> { Internal::Types::Boolean }, optional: true, nullable: false, api_name: "inferAgentBoundaries"

      field :infer_tool_call_links, -> { Internal::Types::Boolean }, optional: true, nullable: false, api_name: "inferToolCallLinks"
    end
  end
end

File: python/src/sikaru_api/types/trace_import_options_request.py

import typing

import pydantic
import typing_extensions
from ..core.pydantic_utilities import IS_PYDANTIC_V2, UniversalBaseModel
from ..core.serialization import FieldMetadata


class TraceImportOptionsRequest(UniversalBaseModel):
    include_feedback: typing_extensions.Annotated[
        typing.Optional[bool], FieldMetadata(alias="includeFeedback"), pydantic.Field(alias="includeFeedback")
    ] = None
    include_pending_spans: typing_extensions.Annotated[
        typing.Optional[bool], FieldMetadata(alias="includePendingSpans"), pydantic.Field(alias="includePendingSpans")
    ] = None
    infer_agent_boundaries: typing_extensions.Annotated[
        typing.Optional[bool], FieldMetadata(alias="inferAgentBoundaries"), pydantic.Field(alias="inferAgentBoundaries")
    ] = None
    infer_tool_call_links: typing_extensions.Annotated[
        typing.Optional[bool], FieldMetadata(alias="inferToolCallLinks"), pydantic.Field(alias="inferToolCallLinks")
    ] = None

    if IS_PYDANTIC_V2:
        model_config: typing.ClassVar[pydantic.ConfigDict] = pydantic.ConfigDict(extra="allow", frozen=True)  # type: ignore # Pydantic v2
    else:

        class Config:
            frozen = True
            smart_union = True
            extra = pydantic.Extra.allow
